### PR TITLE
Clarify tmp_wk and tmp_vort usages in d_sw.py

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

This PR clarifies some of the meaning and dataflow of tmp_wk and tmp_vort, which should enable future performance improvements and stencil merging by decoupling unrelated stencils.

## Code changes:

- added vorticity_agrid and vorticity_bgrid_damped storages to DGridShallowWaterLagrangianDynamics
- added TODOs for optimizations

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] For each public change and fix in `pace-util`, HISTORY has been updated
- [x] Unit tests are added or updated for non-stencil code changes

Additionally, if this PR contains code authored by new contributors:

- [ ] The names of all the new contributors have been added to CONTRIBUTORS.md

### CODE REVIEW

File: fv3core/fv3core/stencils/d_sw.py [hunk 1]
```diff
@@ -1046,12 +1048,18 @@ def __call__(
...
+        # should be mergeable with any compute domain stencil in this object
+
         self._flux_capacitor_stencil(
             cx, cy, mfx, mfy, crx, cry, self._tmp_fx, self._tmp_fy
         )
 
         if not self.hydrostatic:
 
+            # TODO: output value for tmp_wk here is never used, refactor so it is
+            # not unnecessarily computed
```
Comment (elynnwu) [inline fv3core/fv3core/stencils/d_sw.py:1062]: I think the fortran code commonly uses `work`, `wk` or some work combination names to reuse storages. It's great that you're able to decouple this one here. We should probably go through all of them at some point to see if they can all be temporaries, fused into one stencil, or not even compute it (I think sometimes they are calculated to save as diagnostics).

**Review by elynnwu — APPROVED** [review]
lgtm!

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

**Pull request merged**